src/codegen.cpp

```diff
@@ -4421,6 +4451,20 @@ static bool emit_builtin_call(jl_codectx_t &ctx, jl_cgval_t *ret, jl_value_t *f,
         return true;
     }
 
+    else if (f == jl_builtin_memorynew && (nargs == 2)) {
+        const jl_cgval_t &memty = argv[1];
+        if (!memty.constant)
+            return false;
+        jl_datatype_t *typ = (jl_datatype_t*) memty.constant;
+        if (!jl_is_concrete_type((jl_value_t*)typ) || !jl_is_genericmemory_type(typ))
+            return false;
+        jl_genericmemory_t *inst = (jl_genericmemory_t*)((jl_datatype_t*)typ)->instance;
+        if (inst == NULL)
+            return false;
```

Comment:
When does this happen?

Reply:
this means the type isn't `Const` (i.e. for code like
`Memory{rand((Int, Float64))}(undef, 1)`

Reply:
Isn't that the check 3 lines above?

Reply:
oh, right. this case might be for `Memory{1}(undef, 1)` or something like that where it is Const, but not valid.